src/basic/hostname-util.c

```diff
@@ -14,13 +14,25 @@
 #include "strv.h"
 
 char* get_default_hostname(void) {
+        int r;
+
         const char *e = secure_getenv("SYSTEMD_DEFAULT_HOSTNAME");
         if (e) {
                 if (hostname_is_valid(e, 0))
                         return strdup(e);
                 log_debug("Invalid hostname in $SYSTEMD_DEFAULT_HOSTNAME, ignoring: %s", e);
         }
 
+        _cleanup_free_ char *f = NULL;
+        r = parse_os_release(NULL, "DEFAULT_HOSTNAME", &f);
+        if (r < 0)
+                log_debug_errno(r, "Failed to parse os-release, ignoring: %m");
+        if (f) {
```

Comment:
doesn't matter, but you could do `if` → `else if` for an unimportant optimization